spinalcordtoolbox/registration/register.py

```diff
@@ -192,10 +193,19 @@ def register_step_ants_registration(src, dest, step, masking, ants_registration_
     # apply Laplacian filter
     if not step.laplacian == '0':
         logger.info(f"\nApply Laplacian filter")
-        run_proc(['sct_maths', '-i', src, '-laplacian', step.laplacian + ','
-                 + step.laplacian + ',0', '-o', image.add_suffix(src, '_laplacian')])
-        run_proc(['sct_maths', '-i', dest, '-laplacian', step.laplacian + ','
-                 + step.laplacian + ',0', '-o', image.add_suffix(dest, '_laplacian')])
+
+        sigmas = [step.laplacian, step.laplacian, 0]
```

Comment:
Actually, should this step be automatically included in the `maths.laplacian` function so that it's applied by default?

Reply:
Good catch.

> Actually, should this step be automatically included in the `math.laplacian` function so that it's applied by default?

I agree that it would remove some duplication, but from a design point of view I think that the current interface for `def laplacian(data, sigmas)` is correct. Just imagine someone using this API in the future and wonder why all of sudden the user provided standard deviation is modified internally by `math.laplacian()`..

Reply:
> Just imagine someone using this in the future and wonder why all of sudden the user provided standard deviation is modified internally

Good point. Although, now that I think about it, what you describe is kind of already happening on the CLI level, isn't it?

Say the user calls `sct_maths -i t2.nii.gz -laplacian 2.0,3.0,2.0`... those sigma values get modified internally as you describe. :p

Reply:
Thinking about it a little more... I think the step might actually be necessary, and should be included `sct_math.laplacian` like I first suggested. But, this time I have a different reason than code de-duplication.

Since we're specifically a toolbox for spinal cord images, I think `sct_maths -laplacian` is trying to be more than just an invisible wrapper around `ndimage.gaussian_laplace` -- it's specifically tailoring the function to NIFTI images, which have voxel size as part of their spec. So, to ever exclude this adjustment would be applying the wrong laplacian smoothing, I think? It seems mandatory.

My new suggestion would be to have something like this:

```python
def laplacian(img: Image, sigmas: Sequence[float]):
    """
    Apply Laplacian filter to a NIFTI image. 

    :param img: Image object
    :param sigmas: List of floats, one standard deviation value per image dimension. 
    These values are adjusted internally to account for the image voxel size.
    """
    assert len(img.data.shape) == len(sigmas)
    
    # adjust sigma based on voxel size 
    sigmas = [sigmas[i] / img.dim[i + 4] for i in range(len(sigmas))]
    img.data = gaussian_laplace(img.data.astype(float), sigmas)
    return img
```

Otherwise, why should `sct_math.laplacian` exist? Wouldn't the user of the API just use `ndimage.gaussian_laplace` directly?

Reply:
> Otherwise, what's the use of sct_math.laplacian? As it currently is, it's no more useful than just using ndimage.gaussian_laplace directly. :stuck_out_tongue_closed_eyes: 

Yes I agree, my initial reflex was to remove it entirely.

Reply:
After a quick look it seems there are a few more similar cases of seemingly useless wrappers in `math.py` (and probably other easy optimizations). I think I'll save that for a different PR and focus on the subprocess calls for this PR.

Reply:
https://github.com/neuropoly/spinalcordtoolbox/issues/3189